csrc/kernel_cache.cpp

```diff
@@ -1012,7 +1009,9 @@ FusionKernelRuntime::FusionKernelRuntime(
     segmented_fusion_->deserialize(serde_buffer->segmented_fusion());
   }
 
-  heuristics_ = segmented_fusion_->makeInitialHeuristics(args, runtime_info);
+  // Pre-compute the executor order so that the run time path
+  //  would go directly to kernel launch.
+  prepareRuntimeOrder();
```

Comment:
I see this is moved before making initial heuristics. What's the reason for that?

Reply:
In `getMaybeHeuristicsFor`, we're following the `group_run_order` created in `prepareRuntimeOrder` so it needs to occur first before generating the heuristics.

Reference:
```cpp
// Follow group run order
for (int64_t group_id : c10::irange(num_groups)) {
    auto group_to_run = runtime_workspace_.group_run_order.at(group_id);
    ...
}
```